### PR TITLE
Randomness

### PR DESCRIPTION
**Pull request by okhaimie-dev**
**Issue(s):** https://github.com/NethermindEth/StarknetByExample/issues/128

### Description

Added code for Dice Game VRF application.

### Checklist

- [x] **CI Verifier:** Run `./scripts/cairo_programs_verifier.sh` successfully
- [ ] **Contract Tests:** Added tests to cover the changes

### CODE REVIEW

File: src/ch01/dice_game_vrf.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Dice Game using Pragma VRF
+
+This code provides an implementation of a Dice Game contract that utilizes a [Pragma Verifiable Random Function (VRF)](https://docs.pragma.build/Resources/Cairo%201/randomness/randomness) to generate random numbers on-chain.
+
```
Comment (julio4) [inline src/ch01/dice_game_vrf.md:4]: What do you think about explaining a bit more about randomness in the context of blockchain before jumping directly in the implementation? Especially to outline where the entropy comes from and potential risks with it. You can then just link to [Pragma - Randomness](https://docs.pragma.build/Resources/Cairo%201/randomness/randomness#how-randomness-is-generated)

Comment (julio4) [reply src/ch01/dice_game_vrf.md:4]: Entropy sources:
- Blockchain: block hashes, timestamps, ...
- External Sources: Oracles

Risks: Predictability/Manipulation/Centralization-trust

Comment (okhaimie-dev) [reply src/ch01/dice_game_vrf.md:4]: Alright. I'll make an update to this week

**Comment by julio4** [comment]
@okhaimie-dev Hello! Let me know if you need any help on finishing this

**Comment by julio4** [comment]
Done in #238

**Pull request closed without merging**